src/java/org/apache/cassandra/db/compaction/writers/TimeBasedSplittingCompactionWriter.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.db.compaction.writers;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.cassandra.db.*;
+import org.apache.cassandra.db.Directories.DataDirectory;
+import org.apache.cassandra.db.lifecycle.LifecycleTransaction;
+import org.apache.cassandra.db.rows.UnfilteredRowIterator;
+import org.apache.cassandra.io.sstable.Descriptor;
+import org.apache.cassandra.io.sstable.format.SSTableReader;
+import org.apache.cassandra.io.sstable.format.SSTableWriter;
+import org.apache.cassandra.io.sstable.metadata.MetadataCollector;
+
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+public class TimeBasedSplittingCompactionWriter extends CompactionAwareWriter
```

Comment:
To confirm, what this is doing is:
* Split given sstables into new sstables
* How many sstables to create is defined by splitFactor argument
* For given row, we look at which window it falls in (using it's timewindow, the windowSize, the windowStart), and then the sstable. Essentially the duration [windowStart, windowStart + windowSize*windowCount] is partitioned into splitfactor sub-ranger equally.

Assuming my understanding is right, why is windowStart, windowCount manually specified? Isn't it better to look at the data in the sstables and figure out how many windows it covers, and the starting window (the oldest one among all)?

Reply:
yup, something i didnt realise.
Figuring out the range of the windows, is something that is done by another class which creates this class's object.
But yes, it does make sense to put that logic here only.